### DISCUSSION TITLE
The lasso selection corners are too big and other improvements for the selection

### DISCUSSION THREAD

**Discussion by Wladefant** [personalizedrefrigerator/js-draw#38] [Ideas & Feature Requests]
Honestly we are not blind ;). ![SmartSelect_20230925_171703_Joplin (Beta).jpg](https://github.com/personalizedrefrigerator/js-draw/assets/79611093/ad2c5717-558d-4d43-a3fa-ecede377c11c)

And who is ever stretching the edges?

I would suggest something like samsung notes does. 

Firstly change colors of the selection (not so important, your UI is also valid, but I think it would be clearer what you have selected )
Secondly 4 SMALL dots at the end of the corners
![SmartSelect_20230925_172200_Joplin (Beta).jpg](https://github.com/personalizedrefrigerator/js-draw/assets/79611093/6baf0e3c-fc1b-4acb-9eb3-bb80b6e5bd5b)

Or a bit bigger like Microsoft whiteboard does![SmartSelect_20230925_172821_Whiteboard.jpg](https://github.com/personalizedrefrigerator/js-draw/assets/79611093/dfd969ce-f142-482a-8600-acd144834fcd)

Now that I think about it, there are some use cases where that is applicable, then I would suggest something like Microsoft ONENOTE![SmartSelect_20230925_172924_OneNote.jpg](https://github.com/personalizedrefrigerator/js-draw/assets/79611093/e1cb4193-fd73-4dba-a517-f508de923ce1)
And then the rotation button can go into the middle? Or I saw somewhere the function that you can drag the corners or edges in a radius and then it rotates. If you just drag the corners outwards it stretches

The rotation button is good, BUT when I click on the rotation button I except it to rotate 90 degrees to the left.

**Comment by personalizedrefrigerator** [comment]
The rotation button was originally in the center ([see an old release](https://codepen.io/personalizedrefrigerator/pen/MWZGdVP)), but was moved to the top in response to feedback.

Maybe we could do something like Xournal++ and have the rotate button above/to the side of the resize squares:
![screenshot: Xournal++ has a rotate icon to the right of its selection box](https://github.com/personalizedrefrigerator/js-draw/assets/46334387/84a1d34e-346b-4ded-a7d6-a914d83f661a)

Google Docs' drawing feature does something similar:
![Screenshot: Google docs has a rotate icon just above the selection box](https://github.com/personalizedrefrigerator/js-draw/assets/46334387/9f4652e9-4c70-4213-bc5b-25a3ce247786)

I also like OneNote's "<img width="50" alt="move" src="https://github.com/personalizedrefrigerator/js-draw/assets/46334387/03d33958-ad23-4657-9fa7-c1fbcce23000"/>" icon in the center. So maybe we do both.

**Reply by Wladefant** [reply]
oh yes, I was going for google docs. I just forgot how it looked. This round thing is very intuitive. And then the added stroking to the text like onenote or samsungnote and it is perfect.

**Reply by personalizedrefrigerator** [reply]
I plan to work on this soon, but for reference (and if I don't end up implementing this in a timely manner), the relevant files are
- https://github.com/personalizedrefrigerator/js-draw/blob/af1a27f6660ec86ab711131880eb465b7dc52b46/packages/js-draw/src/tools/SelectionTool/Selection.ts#L57 (where the selection handles are created)
- https://github.com/personalizedrefrigerator/js-draw/blob/main/packages/js-draw/src/tools/SelectionTool/SelectionHandle.ts (the selection handles)
- https://github.com/personalizedrefrigerator/js-draw/blob/main/packages/js-draw/src/tools/SelectionTool/SelectionTool.scss (related SCSS styles)

**Reply by personalizedrefrigerator** [reply]
With the latest release, it's a bit closer:
<img alt="screenshot: Main selection handles smaller" src="https://github.com/personalizedrefrigerator/js-draw/assets/46334387/a11e9920-9532-4d1d-8e61-262084e8a9c9" width="440"/>

Changes:
- Decreased visible selection handle size (clickable size is the same)
- Clicking on the <img alt="rotate" width="30" src="https://github.com/personalizedrefrigerator/js-draw/assets/46334387/8937e2dc-d120-47a2-881c-869fb7731a6a"/> handle now rotates by 90 degrees counter-clockwise
- Added additional selection handle

To-do:
- [ ] Add selection handles to other corners
    - [ ] **Fix bug:** resizing (sometimes) happens in the wrong direction when these handles are enabled
- [ ] Move the rotation handle above the selection and connect it to the main selection with a line

**Reply by Wladefant** [reply]
If you could pack the new version into an apk, I would be glad to test. About maintaining: University started and it is much more difficult, than I though haha. I will report back for sure, if it gets easier and I have the time
